Add tensorflow.pad frontend function

div5252 opened the pull request:
Close #7911

div5252 commented:
@CatB1t You can review this PR

CatB1t commented on ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py:
```diff
@@ -875,3 +875,69 @@ def test_tensorflow_gather_nd(
+            min_value=-100,
+            max_value=100,
+        ).filter(lambda x: x[0][0] not in ["float16", "bfloat16"])
+    )
```
We shouldn't be filtering data types, the `helpers.get_dtypes` won't generate invalid dtypes if the function was marked that it doesn't support data type `x`, instead of filtering the data types, you should flag that the function doesn't support the specific data type.

div5252 commented:
I have removed the filtering. tf.pad requires the padding for mode = 'reflect' to be less than ndim - 1, which I have added (https://www.tensorflow.org/api_docs/python/tf/pad).

ivy-seed commented:
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

div5252 commented:
@CatB1t Any further changes required?

fspyridakos commented:
Made some changes to fix some functionality, i.e. mode is case_insensitive in tf but case sensitive in ivy and paddings can be a tensor in tf but must be a list or tuple in ivy.
